envi/archs/rx/disasm.py

```diff
@@ -0,0 +1,1349 @@
+        self.iflags = iflags | envi.ARCH_RX
+        envi.Opcode.__init__(self, va, opcode, mnem, 0, size, operands, iflags)
+
+    def __len__(self):
+        return self.size
+
+
+    def __repr__(self):
+        """
+        Render this opcode to the specified memory canvas
+        """
+        mnem = self.mnem
+        if self.iflags & IF_BYTE:
+            mnem += '.b'
+        elif self.iflags & IF_WORD:
+            mnem += '.w'
+        elif self.iflags & IF_LONG:
+            mnem += '.l'
+        elif self.iflags & IF_UWORD:
+            mnem += '.uw'
+        elif self.iflags & IF_24BIT:
+            mnem += '.a'
+        elif self.iflags & IF_SMALL:
+            mnem += '.s'
+
+
+        return mnem + " " + ",".join([o.repr(self) for o in self.opers])
+
+    def render(self, mcanv):
+        """
+        Render this opcode to the specified memory canvas
+        """
+        mnem = self.mnem
+        if self.iflags & IF_BYTE:
+            mnem += '.b'
+        elif self.iflags & IF_WORD:
+            mnem += '.w'
+        elif self.iflags & IF_LONG:
+            mnem += '.l'
+        elif self.iflags & IF_UWORD:
+            mnem += '.uw'
+
+        mcanv.addNameText(mnem, typename="mnemonic")
+        mcanv.addText(" ")
+
+        # Allow each of our operands to render
+        imax = len(self.opers)
+        lasti = imax - 1
+        for i in range(imax):
+            oper = self.opers[i]
+            oper.render(mcanv, self, i)
+            if i != lasti:
+                mcanv.addText(", ")
+
+    def getBranches(self, emu=None):
+        ret = []
+
+        # To start with we have no flags ( except our arch )
+        flags = self.iflags & envi.ARCH_MASK
+        addb = False
+
+        # If we are a conditional branch, even our fallthrough
+        # case is conditional...
+        if (self.iflags & IF_BRANCH):
+            addb = True
+            if not (self.iflags & IF_NOFALL):
+                flags |= envi.BR_COND
+
+        # If we can fall through, reflect that...
+        if not self.iflags & envi.IF_NOFALL:
+            ret.append((self.va + self.size, flags|envi.BR_FALL))
+
+        # if we have no operands, it has no further branches...
+        if len(self.opers) == 0:
+            return ret
+
+        # Check for a call...
+        if self.iflags & IF_CALL:
+            flags |= envi.BR_PROC
+            addb = True
+
+        if addb:
+            oper0 = self.opers[0]
+            if oper0.isDeref():
+                flags |= envi.BR_DEREF
+                tova = oper0.getOperAddr(self, emu=emu)
+            else:
+                tova = oper0.getOperValue(self, emu=emu)
+
+            ret.append((tova, flags))
+
+        return ret
+
+#FIXME: This should just be wrapped into envi
+def addrToName(mcanv, va):
+    sym = mcanv.syms.getSymByAddr(va)
+    if sym is not None:
+        return repr(sym)
+    return "0x%.4x" % va
+
+def renderPossibleLocation(mcanv, op, idx, value):
```

Comment:
So I know the msp430 module defines something like this too, but nobody actually calls this?

Reply:
MSP430 uses it, RX does not.  removed.